src/charm.py

```diff
@@ -1,115 +1,117 @@
-        if log_level in VALID_LOG_LEVELS:
-            # The config is good, so update the configuration of the workload
-            container = self.unit.get_container("httpbin")
-            # Verify that we can connect to the Pebble API in the workload container
-            if container.can_connect():
-                # Push an updated layer with the new config
-                container.add_layer("httpbin", self._pebble_layer, combine=True)
-                container.replan()
-
-                logger.debug("Log level for gunicorn changed to '%s'", log_level)
-                self.unit.status = ops.ActiveStatus()
-            else:
-                # We were unable to connect to the Pebble API, so we defer this event
-                event.defer()
-                self.unit.status = ops.WaitingStatus("waiting for Pebble API")
-        else:
-            # In this case, the config option is bad, so block the charm and notify the operator.
-            self.unit.status = ops.BlockedStatus("invalid log level: '{log_level}'")
-
-    @property
-    def _pebble_layer(self):
-        """Return a dictionary representing a Pebble layer."""
-        return {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {
-                        "GUNICORN_CMD_ARGS": f"--log-level {self.model.config['log-level']}"
-                    },
-                }
-            },
-        }
-
-
-if __name__ == "__main__":  # pragma: nocover
-    ops.main.main(IsCharmsTemplateCharm)
+        if not self.model.get_relation(AGENT_RELATION):
+            self.model.unit.status = ops.BlockedStatus("Waiting for relation.")
+            return
+
+        if not self.state.agent_relation_credentials:
+            self.model.unit.status = ops.WaitingStatus("Waiting for complete relation data.")
+            logger.info("Waiting for complete relation data.")
+            return
+
+        self.model.unit.status = ops.MaintenanceStatus("Starting agent service.")
+        try:
+            self.jenkins_agent_service.restart()
+        except service.ServiceRestartError as exc:
+            logger.error("Error restarting the agent service %s", exc)
+            raise RuntimeError("Error restarting the agent service") from exc
+
+        self.model.unit.status = ops.ActiveStatus()
+
+    def _on_update_status(self, _: ops.UpdateStatusEvent) -> None:
+        """Check status of the charm and report back to juju."""
+        logger.debug(
+            "Jenkins agent service is currently up: %s", self.jenkins_agent_service.is_active
+        )
+        if self.jenkins_agent_service.is_active:
```

Comment:
I don't see across the code under which circumstances `self.jenkins_agent_service.is_active`might be true while the unit is not active. Is this really needed?

Reply:
Thanks, initially I thought to use this to consolidate the status of the service and the status of the charm, but now with proper error handling and the active status is also set at the end of the restart method, this is redundant. I've removed it.